### ISSUE TITLE
Locator.MAXTICKS exceeded for very errant GPS locations

### ISSUE THREAD

**Issue by liamtoney** [uafgeotools/cube_conversion#5]
John Lyons reports the following error message:
```
Coordinates exported to HV.FIS8.02.DDF.json
Traceback (most recent call last):
  File "/home/jlyons/Python/cubeconversion/cube_convert.py", line 445, in <module>
    png_filename = json_filename.rstrip('.json') + '.png'
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/figure.py", line 2180, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/backends/backend_qt5agg.py", line 88, in print_figure
    super().print_figure(*args, **kwargs)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/backend_bases.py", line 2056, in print_figure
    **kwargs)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 527, in print_png
    FigureCanvasAgg.draw(self)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py", line 388, in draw
    self.figure.draw(self.renderer)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/artist.py", line 38, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/figure.py", line 1709, in draw
    renderer, self, artists, self.suppressComposite)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/image.py", line 135, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/artist.py", line 38, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/axes/_base.py", line 2645, in draw
    mimage._draw_list_compositing_images(renderer, self, artists)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/image.py", line 135, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/artist.py", line 38, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/axis.py", line 1204, in draw
    ticks_to_draw = self._update_ticks()
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/axis.py", line 1080, in _update_ticks
    major_locs = self.get_majorticklocs()
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/axis.py", line 1325, in get_majorticklocs
    return self.major.locator()
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/ticker.py", line 1799, in __call__
    return self.tick_values(vmin, vmax)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/ticker.py", line 1808, in tick_values
    return self.raise_if_exceeds(locs)
  File "/home/jlyons/anaconda3/lib/python3.7/site-packages/matplotlib/ticker.py", line 1520, in raise_if_exceeds
    len(locs), locs[0], locs[-1]))
RuntimeError: Locator attempting to generate 8471 ticks from -39970.0 to 2380.0: exceeds Locator.MAXTICKS
```
The 2-D histogram plotting code places major ticks every 5 m, with minor ticks every 1 m. Usually this results in a reasonable number of ticks. However in this case, the range of one of the axes was "-39970.0 to 2380.0" m which is > 42 km! Probably related to a really poor GPS lock at one point.

Enforcing maximum x- and y-axis limits will likely be a sufficient solution for this error.

**Comment by davidfee5** [comment]
I know I've already said this, but couldn't doing a 95% trimmed median/mean only help?

**Comment by liamtoney** [comment]
Bug fixed in d3a3713. Since we've decided to use the "mode" coordinate, trimming doesn't matter... but we can/should revisit this later if we run into weird cases.
